test/reviewController.test.js

```diff
@@ -0,0 +1,113 @@
+const { app, connectDB } = require('../src/expressServer'); // Import your Express app here
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const Review = require('../src/models/Review');
+const User = require('../src/models/User');
+const Album = require('../src/models/Album');
+
+let server;
+let mongooseConnection;
+let mongodb;
+
+beforeAll(async () => {
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('ReviewController API Tests', () => {
+  beforeAll(async () => {
+    await Review.deleteMany({});
+    await User.create({
+      email: 'Emily@google.com',
+      password: 'secret',
+      name: 'Emily',
+      username: 'emily123',
+      role: 'user',
+    });
+  });
+
+  beforeEach(async () => {
+    await Review.deleteMany({});
+    // Create a unique album and user for each test
+    const album = await Album.create({
+      albumName: 'Unique Album',
+      artistName: 'Unique Artist',
+      spotifyUrl: 'https://api.spotify.com/v1/albums/unique',
+    });
+
+    const user = await User.findOne({ username: 'emily123' });
```

Comment:
Instead of setting the `user` variable here by fetching from the database, we can set the variable earlier on in the `beforeAll` block when we create the user. 

On line 30; we can do `user = await User.create(...)`

---

Though to be fair, in that case, we'd need to declare the `user` variable with `let user` somewhere outside of both blocks so that both blocks have access to that variable (it's "in scope" for both blocks). This could be for example, at the top of the `describe` block (between line 27 and 28)  or outside of the `describe` block like where we declare our server, mongodb, and mongooseConnection variables.